tlatools/org.lamport.tlatools/src/tla2sany/semantic/Generator.java

```diff
@@ -1947,20 +1943,22 @@ public final SymbolTable getSymbolTable() {
 		return symbolTable;
 	}
 
-	public final ModuleNode generate(TreeNode treeNode) throws AbortException {
+	public final Optional<ModuleNode> generate(TreeNode treeNode) throws AbortException {
```

Comment:
Change of api. I can also create a new method and mark the old one as deprecated.